nx-X11/include/Xos_r.h

```diff
@@ -619,36 +587,11 @@ typedef struct {
 # endif
 } _Xreaddirparams;
 
-# if defined(_POSIX_THREAD_SAFE_FUNCTIONS) || defined(AIXV3) || \
-     defined(AIXV4) || defined(__APPLE__) || defined(__DARWIN__)
-/* AIX defines the draft POSIX symbol, but uses the final API. */
+# if defined(_POSIX_THREAD_SAFE_FUNCTIONS) || \
+     defined(__APPLE__) || defined(__DARWIN__)
 /* POSIX final API, returns (int)0 on success. */
-#  if defined(__osf__)
-/* OSF/1 V4.0 <dirent.h> doesn't declare _Preaddir_r, breaking under C++. */
-extern int _Preaddir_r(DIR *, struct dirent *, struct dirent **);
-#  endif
 #  define _XReaddir(d,p)						\
     (readdir_r((d), &((p).dir_entry), &((p).result)) ? NULL : (p).result)
-# elif defined(_POSIX_REENTRANT_FUNCTIONS) && defined(__osf__)
```

Comment:
This can stay deleted up to `# elif defined(_POSIX_REENTRANT_FUNCTIONS)` on line 648, because `__osf__` is gone.